src/porepy/viz/diagnostics_mixin.py

```diff
@@ -0,0 +1,383 @@
+                "big matrices. It is recommended to reduce the problem size or to use "
+                'the option "is_plotting_max" only.'
+            )
+
+        # Determining the block indices and collecting the submatrices.
+        equation_indices = self._equations_indices(grouping=grouping)
+        variable_indices = self._variable_indices(grouping=grouping)
+
+        if grouping is None and max(len(equation_indices), len(variable_indices)) > 10:
+            logger.warning(
+                "Treating all the subdomains separately might not be "
+                "informative if there are many of them. It is recommended to set"
+                "\"grouping='subdomains'\""
+            )
+
+        self._collect_submatrices(
+            mat=full_matrix,
+            equation_indices=equation_indices,
+            variable_indices=variable_indices,
+        )
+
+        # Computing the required features of each block.
+        block_matrix_shape = len(equation_indices), len(variable_indices)
+        block_data: dict[tuple[int, int], dict] = defaultdict(dict)
+        for i, j in product(range(block_matrix_shape[0]), range(block_matrix_shape[1])):
+            submat = self.submatrices[i, j]
+            is_empty_block = submat.data[submat.data != 0].size == 0
+            block_data[i, j]["is_empty_block"] = is_empty_block
+            if not is_empty_block:
+                if is_plotting_condition_number:
+                    block_data[i, j]["condition_number"] = get_condition_number(submat)
+                if is_plotting_max:
+                    block_data[i, j]["max"] = abs(submat.data).max()
+
+        # Plotting the figures.
+        if is_plotting_condition_number:
+            _plot_condition_number(
+                block_data,
+                variable_names=tuple(variable_indices.keys()),
+                equation_names=tuple(equation_indices.keys()),
+            )
+            plt.show()
+
+        if is_plotting_max:
+            _plot_max(
+                block_data,
+                variable_names=variable_indices.keys(),
+                equation_names=equation_indices.keys(),
+            )
+            plt.show()
+
+    def _equations_indices(
+        self, grouping: Optional[Literal["subdomains"]] = None
+    ) -> dict[str, np.ndarray]:
+        """Collects the indices of equations presented in the Jacobian matrix as rows.
+
+        Returns:
+            A dictionary with keys - printed names of the equations, values - indices of
+            this equation in the global Jacobian matrix.
+        """
+        equation_indices = {}
+        assert grouping in (None, "subdomains"), f"Unknown grouping: {grouping}"
+
+        assembled_equation_indices = self.equation_system.assembled_equation_indices
+
+        if grouping is None:
+            block_indices = 0
+            for eq_name, dof_indices in assembled_equation_indices.items():
+                equation_image_space = (
```

Comment:
I am struggling to see the ordering of the dof indices for a random subdomain here. There is no orderign in the loop, which means the order will be the same as in equation_image_space (since dictionaries preserve order). I guess this is okay, but have you thought through this?